utils/mail/mailer.js

```diff
@@ -157,4 +157,53 @@ export default class NewsletterMailer {
             auth: {user: mailConfigs.auth.user, pass: mailConfigs.auth.pass}
         });
     }
+
+    /**
+     * Creates batches of given subscribers.
+     *
+     * @param {Subscriber[]} subscribers - The array of subscribers to be batched.
+     * @param {number} batchSize - The number of subscribers in each batch.
+     *
+     * @returns {Subscriber[][]} An array of subscriber arrays, where each inner array is a batch.
+     */
+    #createBatches(subscribers, batchSize) {
+        const batches = [];
+        for (let i = 0; i < subscribers.length; i += batchSize) {
+            batches.push(subscribers.slice(i, i + batchSize));
+        }
+
+        logDebug(logTags.Newsletter, `Created ${batches.length} batches.`);
+        return batches;
+    }
+
+    /**
+     * Send emails in batches with appropriate delay.
+     *
+     * @returns {Promise<number>} Total emails sent.
+     */
+    async #processBatches(transporter, mailConfig, batches, chunkSize, tierIds, post, fullContent, partialContent, unsubscribeLink, delayBetweenBatches) {
```

Comment:
We can remove the transporter parameter here and build one directly as we have access to the required `mailConfig` object.

Reply:
On the contrary, we don't need this even here. Instead a transporter can directly be created in the `#sendEmailToSubscriber` method.

Reply:
Done.